tgui/packages/tgui/interfaces/PowerMonitor/Apc.tsx

```diff
@@ -81,48 +83,112 @@ export const PowerMonitorApcGlobal = (_props, context) => {
       </Tooltip>
       <Tooltip content="Lighting">
         <Table.Cell header collapsing>
-          Lgt.
+          <Header
+            sortDirection={props.state?.field === ApcTableHeaderColumns.Lighting ? props.state.dir : null}
+            onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Lighting)}>
+            Lgt.
+          </Header>
         </Table.Cell>
       </Tooltip>
       <Tooltip content="Environment">
         <Table.Cell header collapsing>
-          Env.
+          <Header
+            sortDirection={props.state?.field === ApcTableHeaderColumns.Environment ? props.state.dir : null}
+            onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Environment)}>
+            Env.
+          </Header>
         </Table.Cell>
       </Tooltip>
       <Table.Cell header textAlign="right">
-        Load
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.Load ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Load)}>
+          Load
+        </Header>
       </Table.Cell>
       <Table.Cell header textAlign="right">
-        Cell Charge
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.CellCharge ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.CellCharge)}>
+          Cell Charge
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.CellState ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.CellState)}>
+          Cell State
+        </Header>
       </Table.Cell>
-      <Table.Cell header>Cell State</Table.Cell>
     </>
   );
 };
 
+const apcDataComparer
+  = (a: PowerMonitorApcItemData,
+    b: PowerMonitorApcItemData,
+    names: Record<string, string>,
+    field: ApcTableHeaderColumns): number => {
+
+    if (field === ApcTableHeaderColumns.Area) {
+      return names[a[field]].localeCompare(names[b[field]]);
+    } else {
+      return numericCompare(a[field], (b[field]));
+    }
+  };
+
+const sortApcData
+  = (data: PowerMonitorApcItemData[],
+    names: Record<string, string>,
+    state: ApcTableHeaderColumnSortState): PowerMonitorApcItemData[] => {
+
+    if (state !== null) {
+      data.sort((a, b) => apcDataComparer(a, b, names, state.field));
+      if (state.dir === SortDirection.Asc) {
+        data.reverse();
+      }
+    }
+
+
+    return data;
+  };
+
 type PowerMonitorApcTableRowsProps = {
   search: string;
+  sortState: ApcTableHeaderColumnSortState;
 };
-
 export const PowerMonitorApcTableRows = (props: PowerMonitorApcTableRowsProps, context) => {
   const { search } = props;
   const { data } = useBackend<PowerMonitorApcData>(context);
 
   return (
     <>
-      {data.apcs.map((apc) => (
+      {sortApcData(data.apcs, data.apcNames, props.sortState).map((apc) => (
```

Comment:
`props` is deconstructed above, better to keep to that pattern (i.e. deconstruct props (const `{ foo } = props;`) vs. access as `props.foo`). No real issue with use of `props.foo`, but blending the two makes tracking down props usage annoying.

Can do the same with `data`, but as you're accessing them all the same way (as `data.foo`) I have no issue with it. Essentially try to keep to one method of nested access for `props` and one for `data` (can obviously be the same method, but doesn't have to be).

Reply:
Not a fan of the deconsturcting props pattern in general, but do agree that one or the other should be used, at least witihn the same file for consistancy.